rclcpp_components/src/component_manager.cpp

```diff
@@ -178,6 +178,17 @@ ComponentManager::OnLoadNode(
                     "Extra component argument 'use_intra_process_comms' must be a boolean");
           }
           options.use_intra_process_comms(extra_argument.get_value<bool>());
+        } else if (extra_argument.get_name() == "domain") {
+          if (extra_argument.get_type() != rclcpp::ParameterType::PARAMETER_INTEGER) {
+            throw ComponentManagerException(
+                    "Extra component argument 'domain' must be an integer");
+          }
+          std::size_t domain_id = extra_argument.get_value<std::size_t>();
```

Comment:
At this point, the extra_argument is an integer (a signed type) and is then being returned as a size_t (unsigned). Negative values would get through the first check and then would be converted to large positive values. Depending on how careful one wants to be, there could be an additional semantic check on the domain_id's value before the conversion.

Reply:
Thanks for pointing that out, I've addressed it in: https://github.com/ros2/domain_bridge/pull/32/commits/c3bbf8e94b3a48ecb30568f3c0728002b3d9a6c8